leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceService.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Bosch Software Innovations GmbH - initial API
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue;
+
+import org.eclipse.leshan.server.registration.Registration;
+
+/**
+ * Tracks the status of each LWM2M client registered with Queue mode binding. Also ensures that the
+ * {@link PresenceListener} are notified on state changes only for those LWM2M clients registered using Queue mode
+ * binding.
+ *
+ * @see Presence
+ */
+public interface PresenceService {
+
+    /**
+     * sets the client identified by the given registration in {@link Presence#OFFLINE} mode if the current state is
+     * either {@link Presence#ONLINE}. If the state is changed successfully, then the corresponding
+     * {@link PresenceListener} are notified.
+     * 
+     * @param endpoint client
+     */
+    public void setOffline(Registration registration);

```

Comment:
I'm not sure we want to expose `setOffline` and `setOnline` in public API ? Should we just keep this in the implementation ?